pkg/querier/queryrange/split_by_interval.go

```diff
@@ -338,8 +342,8 @@ func maxRangeVectorAndOffsetDuration(q string) (time.Duration, time.Duration, er
 
 // reduceSplitIntervalForRangeVector reduces the split interval for a range query based on the duration of the range vector.
 // Large range vector durations will not be split into smaller intervals because it can cause the queries to be slow by over-processing data.
-func reduceSplitIntervalForRangeVector(r queryrangebase.Request, interval time.Duration) (time.Duration, error) {
-	maxRange, _, err := maxRangeVectorAndOffsetDuration(r.GetQuery())
+func reduceSplitIntervalForRangeVector(r *LokiRequest, interval time.Duration) (time.Duration, error) {
+	maxRange, _, err := maxRangeVectorAndOffsetDuration(r.Plan.AST)
```

Comment:
`r` is a pointer type so we should guard against panics here

Reply:
Hm, this is not done in any other method that accepts `queryrangebase.Request` etc. It depends on the contracts these methods have. I think `nil` for a request is a bug that shows way up and it should indeed panic and result in an HTTP 500 instead of an error.

Happy to add the check though.

Reply:
Shouldn't we avoid getting to the point that we call this function though, if the query couldn't be parsed into a plan properly?

Reply:
Yes. That's why I think it should [fail loud and proud](https://github.com/mesosphere/marathon/blob/90619da70d79a2ecf19368a21c6a695ea981bd74/docs/docs/code-culture.md#fail-loud-and-proud), ie panic. To be fair, an error is not too different but it's not loud enough for me 🙂